packages/web3-shared/base/src/specs/index.ts

```diff
@@ -549,6 +549,8 @@ export interface TransactionDescriptor<ChainId, Transaction> {
     tokenInAddress?: string
     /** The amount of the token leveraged to swap other tokens */
     tokenInAmount?: string
+    /** The custom token description */    
+    tokenDescription?: string
```

Comment:
A descriptor will need to keep itself as generic as possible. The `tokenInAddress` and `tokenInAmount`  are related to a swapping transaction, not all transactions. The same problem is if we add a `tokenDescription`.

Reply:
If a transaction has an input token, an output token could also exist. Then it also makes sense when we talk about the token description.